text_handler.py

```diff
@@ -26,42 +29,58 @@ async def __handleYoutubeDownload(self, message: types.Message):
+                    'format': 'bestaudio',
+            }
+            with YoutubeDL(ydl_opts) as ydl:
+                info = ydl.extract_info(message.text, download=False)
+                audio_url = format(info['url'])
             telegram_audio_limit = 52428800
+            request_head = Request(audio_url, method='HEAD')
+            audio_filesize = urlopen(request_head).headers['Content-Length']
 
-            if (filesize < telegram_audio_limit and length < 600):
-                output_audiofile = stream.download(output_path="downloads/",
-                                                   filename=title +
-                                                   " - YouTube - " +
-                                                   video_id)
-                output_thumbnail = "downloads/square_thumbnail_" + video_id + \
-                                   ".jpg"
-                self.__make_squarethumb(yt.thumbnail_url, output_thumbnail)
+            if int(audio_filesize) < telegram_audio_limit:
+                ydl.download([message.text])
+                duration = format(info['duration'])
+                extractor = format(info['extractor'])
+                title = format(info['title'])
+                audio_ext = format(info['ext'])
+                webpage_id = format(info['id'])
+                webpage_url = format(info['webpage_url'])
+                uploader = format(info['uploader'])
+                thumbnail_url = format(info['thumbnail'])
+                thumbnail_urlpath = urlparse(thumbnail_url).path
+                thumbnail_filename = os.path.basename(thumbnail_urlpath)
+                thumbnail_ext = thumbnail_filename.split(".")[-1]
+                base_filename = "downloads/" + extractor + '_' + \
+                    webpage_id + '.'
+                output_audiofile = base_filename + audio_ext
+                original_thumbnail = base_filename + thumbnail_ext
+                output_thumbnail = "downloads/square_thumbnail_" + \
+                    webpage_id + ".jpg"
+                self.__make_squarethumb(original_thumbnail, output_thumbnail)
                 with open(output_audiofile, "rb") as audio, \
                      open(output_thumbnail, "rb") as thumb:
                     await message.reply_audio(audio,
-                                              caption="<a href=\"" + video_url +
-                                                      "\">" + title + "</a>",
+                                              caption="<b><a href=\"" +
+                                              webpage_url + "\">" +
+                                              title + "</a></b>",
                                               parse_mode=ParseMode.HTML,
-                                              duration=length,
-                                              performer=author,
+                                              duration=duration,
```

Comment:
looks like all the fields of **info[]** are **Strings**, but here for **duration** you're expecting an **Integer**, so type checker is not happy about it
<img width="728" alt="Screenshot 2020-12-07 at 00 59 47" src="https://user-images.githubusercontent.com/49668442/101288157-99ca8b00-3827-11eb-9038-d6826eda5a9f.png">

if this string is a proper number i suggest to convert it to an integer like this:
`duration=int(duration),`

Reply:
didn't notice that before, btw on youtube the number is rounded, but on soundcloud it's not, usually something like `267.341`, so I think it should be `duration=int(round(duration))` here

Reply:
yea you need to experiment witht that, i think if you have a string ` duration = "23.123"` and you'll try to do `int(duration)` then it'll remove the digits after `.`, not too sure how it works on python